samples/azure-quantum/grover/Grover.ipynb

```diff
@@ -0,0 +1,710 @@
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "Next, define the diffusion operator. This reflects the input qubits about the uniform superposition state."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[\"ReflectAboutUniform\"]",
+      "text/html": [
+       "<ul><li>ReflectAboutUniform</li></ul>"
+      ],
+      "text/plain": [
+       "ReflectAboutUniform"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "operation ReflectAboutUniform(inputQubits : Qubit[]) : Unit {\n",
+    "    within {\n",
+    "        // Transform the uniform superposition to all-zero.\n",
+    "        Adjoint PrepareUniform(inputQubits);\n",
+    "        // Transform the all-zero state to all-ones\n",
+    "        PrepareAllOnes(inputQubits);\n",
+    "    } apply {\n",
+    "        // Now that we've transformed the uniform superposition to the\n",
+    "        // all-ones state, reflect about the all-ones state, then let\n",
+    "        // the within/apply block transform us back.\n",
+    "        ReflectAboutAllOnes(inputQubits);\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "## Grover's operation\n",
+    "Now we write a Q# operation for Grover's search algorithm. Start by defining an function that will return the number of Grover iterations needed to find a single marked item, given the number of qubits in a register. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[\"NIterations\"]",
+      "text/html": [
+       "<ul><li>NIterations</li></ul>"
+      ],
+      "text/plain": [
+       "NIterations"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "open Microsoft.Quantum.Math;\n",
+    "open Microsoft.Quantum.Convert;\n",
+    "\n",
+    "function NIterations(nQubits : Int) : Int {\n",
+    "    let nItems = 1 <<< nQubits; // 2^numQubits\n",
+    "    // compute number of iterations:\n",
+    "    let angle = ArcSin(1. / Sqrt(IntAsDouble(nItems))); //arcsin(1 / sqrt(2^numQubits))\n",
+    "    let nIterations = Round(0.25 * PI() / angle - 0.5); // (pi/4) / arcsin(1 / sqrt(2^numQubits)) - 0.5\n",
```

Comment:
Similarly here, I'm not sure the comment adds anything in this case?

Reply:
Removed this comment as well